In [86]:
#Importing the necessary libraries

import seaborn as sns
from sklearn.model_selection import train_test_split
from sklearn.metrics import accuracy_score, confusion_matrix, recall_score
from sklearn.ensemble import RandomForestClassifier
from scipy import stats

In [69]:
#Loading the first five rows of the dataset

df = sns.load_dataset('iris')
df.head()

sepal_length  sepal_width  petal_length  petal_width species
0           5.1          3.5           1.4          0.2  setosa
1           4.9          3.0           1.4          0.2  setosa
2           4.7          3.2           1.3          0.2  setosa
3           4.6          3.1           1.5          0.2  setosa
4           5.0          3.6           1.4          0.2  setosa

In [70]:
#Getting the sahpe of the dataset

df.shape

(150, 5)

In [71]:
#Finding null values

df.isna().sum()

sepal_length    0
sepal_width     0
petal_length    0
petal_width     0
species         0
dtype: int64

In [72]:
#finding duplicates

df.duplicated().sum()

1

In [73]:
#Dropping duplicates

df.drop_duplicates(inplace = True)
df.duplicated().sum()

0

In [74]:
df['sepal_width1'] = stats.zscore(df['sepal_width'])
cleaned_df = df[(df['sepal_width1'] >= -3) & (df['sepal_width1'] <= 3)]
print(cleaned_df.shape)

(148, 6)


In [75]:
#dropping the new column

cleaned_df.drop(columns='sepal_width1', inplace=True)
cleaned_df.shape

C:\Users\USER\AppData\Local\Temp\ipykernel_12756\1942551411.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  cleaned_df.drop(columns='sepal_width1', inplace=True)


(148, 5)

In [76]:
#Converting the species column to codes

cleaned_df['species'] = cleaned_df['species'].astype('category').cat.codes

cleaned_df['species'].unique()

C:\Users\USER\AppData\Local\Temp\ipykernel_12756\2705071763.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  cleaned_df['species'] = cleaned_df['species'].astype('category').cat.codes


array([0, 1, 2], dtype=int8)

In [89]:
#Feature engineering

x = cleaned_df[['sepal_length','sepal_width','petal_length','petal_width']]
y = cleaned_df['species']

#Splitting the dataset

x_train,x_test,y_train,y_test = train_test_split(x,y,test_size=0.2,random_state=42)

#Selecting the model

model = RandomForestClassifier()

#Fitting and training the model

model.fit(x_train, y_train)

#Making predictions

y_predict = model.predict(x_test)

#Checking the accuracy of the model

accuracy = accuracy_score(y_predict, y_test)
print('Accuracy of the model = ', round(accuracy*100, 2), '%')


#Getting a confusion matrix

matrix = confusion_matrix(y_predict, y_test)
print('\nThe confusion matrix of the model is: \n',matrix)

Accuracy of the model =  83.33 %

The confusion matrix of the model is: 
 [[11  0  0]
 [ 0  7  2]
 [ 0  3  7]]


The model that we developed has a 83% chance of predicting the correct iris flower type. 